nitro-protocol/src/contract/multi-asset-holder.ts

```diff
@@ -96,13 +96,13 @@ export function computeReclaimEffects(
     };
 
     // copy each element except the indexOfTargetInSource element
-    if (sourceAllocations[i].destination == left) {
+    if (sourceAllocations[i].destination.toLowerCase() == left.toLowerCase()) {
```

Comment:
Perhaps... how could we achieve that?

In my view, as long as we are representing bytes as hex strings we need to be careful when doing comparisons like this. I think it would be difficult / dangerous to try and keep them e.g. lowercase everywhere and then rely on a simple `==` check. 

We could introduce a helper `areEqualDestinations()`, but I fear these kinds of bugs will likely still crop up.